### DISCUSSION TITLE
Why does `migrate create <name>` runs the migration created automatically?

### DISCUSSION THREAD

**Discussion by Odas0R** [pocketbase/pocketbase#4604] [Q&A] [answered]
After `migrate create <name>` it creates the migration file, then I add the custom code for the migration.

After that I do `migrate up` to execute the sql, but I get:

![image](https://github.com/pocketbase/pocketbase/assets/32167770/65376f9c-aed2-4d1e-9c35-0aa9fd50a052)

Am I missing something here?

Here's the `MustRegister`: 
```go
AUTO_MIGRATE := os.Getenv("AUTO_MIGRATE")
fmt.Println("AUTO_MIGRATE", AUTO_MIGRATE)
migratecmd.MustRegister(app, app.RootCmd, migratecmd.Config{
	Automigrate: AUTO_MIGRATE == "true",
})
```

`AUTO_MIGRATE` is true when I run `serve`. Maybe because it is always true, when I create a new migration it automatically runs it?

## Fixing it

To fix this I do `migrate down 1`, then `migrate up 1`

**Comment by ganigeorgiev** [comment]
I'm not sure I follow. It is not clear what is the order of the commands you are executing, nor what is your setup.

Even if `Automigrate` is true the created migration will no be applied automatically after `migrate create` (_the `Automigrate` option is usually intended for the cases when doing collection changes from the Admin UI; these **autocreated** migrations are automatically registered because they are generated **after** the db changes have been already applied, eg. after collection change, but that shouldn't matter for `migrate create`_).

Migrations are executed on `migrate up` or on `serve`.

If that's not what you are asking please elaborate and/or provide a clear reproducible steps.

**Reply by ganigeorgiev** [reply]
Also make sure that your created migrations are actually loaded, aka. that you have `_ "yourpackage/migrations"` import somewhere in your main file.

Additionally, if you want to inspect the applied migrations, you can run:

```sh
sqlite3 pb_data/data.db 'select * from _migrations'
```

**Reply by Odas0R** [reply]
https://github.com/pocketbase/pocketbase/assets/32167770/80cd6176-ae1b-4fca-a2f7-f21e0e6862f0


Here's an video demonstrating it.

To actually run a migration after creating it, I always need to `migrate down 1` and then `migrate up`

The `main` binary is bundled pocketbase instance, I'm under a docker container in alpine.

**Reply by ganigeorgiev** [reply]
The above video is not playable for me. In any case, I've just tried to reproduce it locally and it works correctly for me - aka. migrations created with `migrate create` are applied only when running `migrate up` or on `serve`.

**Reply by Odas0R** [reply]
I've fixed the video. Apparently GH doesn't like mp4's.

**Reply by Odas0R** [reply]
![image](https://github.com/pocketbase/pocketbase/assets/32167770/f0a23ede-d0a7-41a6-aa6c-5ac417a977b2)

Here's some more commands:

1. `./main migrate create testing_wow` -> creates
2. `./main migrate down 1` -> shows that `testing_wow` was already applied on create

**Reply by Odas0R** [reply]
OH WAIT. Entirely my bad.

**Comment by Odas0R** [comment, marked as answer]
Entirely my bad, I think I know what it is. I'm using https://github.com/cosmtrek/air, which re-builds the binary on file-changes, and the folder migrations is being listened to, so on create it will re-build the pocketbase binary and then apply the migration, because as you say migrations are applied on serve and I'm re-running `serve` on file-changes.......

Sorry for wasting your time.
